File: src/main/java/space/iseki/cmdpipe/ExecutionInfo.java
package space.iseki.cmdpipe;

import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.time.Instant;

public record ExecutionInfo(
        long pid,
        @NonNull Instant startAt,
        @Nullable Instant endAt,
        @Nullable Integer exitCode,
        boolean timeoutToKilled,
        @NonNull String stderrSnapshot
){}

File: src/main/kotlin/space/iseki/cmdpipe/ExecutionInfo.kt
package space.iseki.cmdpipe

import java.time.Instant

@JvmName("-executionInfoOf")
internal fun executionInfoOf(
    pid: Long,
    startAt: Instant,
    endAt: Instant? = null,
    exitCode: Int? = null,
    timeoutToKilled: Boolean = false,
    stderrSnapshot: String = "",
) = ExecutionInfo(pid, startAt, endAt, exitCode, timeoutToKilled, stderrSnapshot)
